crates/aptos/src/common/utils.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{CliResult, Error};
+use serde::Serialize;
+
+/// Prompts for confirmation until a yes or no is given explicitly
+/// TODO: Capture interrupts
+pub fn prompt_yes(prompt: &str) -> bool {
+    let mut result: Result<bool, ()> = Err(());
+
+    // Read input until a yes or a no is given
+    while result.is_err() {
+        println!("{} [y/n] >", prompt);
+        let mut input = String::new();
+        if std::io::stdin().read_line(&mut input).is_err() {
+            continue;
+        }
+        result = match input.trim().to_lowercase().as_str() {
```

Comment:
love this, but it isn't compatible with TF :P

Reply:
Are we trying to build this around TF?

If you notice, I just built in an override for that in the one use case (`-o` overrides explicitly)